File: problems/intermediate/organizing_container/organizing_container.py

def organizingContainers(container):
    n = len(container)

    # Total number of balls in each container (row sums)
    container_sums = [sum(row) for row in container]

    # Total number of each type of ball (column sums)
    type_sums = [sum(container[i][j] for i in range(n)) for j in range(n)]

    # Sort and compare the two lists
    if sorted(container_sums) == sorted(type_sums):
        return "Possible"
    else:
        return "Impossible"

File: problems/intermediate/organizing_container/organizing_container.js
function organizingContainers(container) {
    const n = container.length;
    const containerSums = new Array(n).fill(0);
    const typeSums = new Array(n).fill(0);

    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            containerSums[i] += container[i][j]; // total balls in container i
            typeSums[j] += container[i][j];      // total balls of type j
        }
    }

    containerSums.sort((a, b) => a - b);
    typeSums.sort((a, b) => a - b);

    for (let i = 0; i < n; i++) {
        if (containerSums[i] !== typeSums[i]) {
            return "Impossible";
        }
    }

    return "Possible";
}

module.exports = { organizingContainers };



